hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/v2/pages/buckets/buckets.tsx

```diff
@@ -0,0 +1,563 @@
+      // e.g. After page is reloaded / redirect from other page
+      COLUMNS[COLUMNS.length - 1] = aclLinkColumn;
+    }
+
+    if (defaultColumns.length > 0 && defaultColumns[defaultColumns.length - 1].label !== 'acls') {
+      defaultColumns.push({
+        label: aclLinkColumn.title as string,
+        value: aclLinkColumn.key as string
+      });
+    }
+  };
+
+  function handleColumnChange(selected: ValueType<Option, true>) {
+    setSelectedColumns(selected as Option[]);
+  }
+
+  function handleLimitChange(selected: ValueType<Option, false>) {
+    setSelectedLimit(selected as Option);
+  }
+
+  const loadData = () => {
+    setLoading(true);
+    const { request, controller } = AxiosGetHelper(
+      '/api/v1/buckets',
+      cancelSignal,
+      '',
+      { limit: selectedLimit.value }
+    );
+    cancelSignal = controller;
+    request.then(response => {
+      const bucketsResponse: BucketResponse = response.data;
+      const totalCount = bucketsResponse.totalCount;
+      const buckets: Bucket[] = bucketsResponse.buckets;
+
+      const dataSource: Bucket[] = buckets.map(bucket => {
+        return {
+          volumeName: bucket.volumeName,
+          name: bucket.name,
+          versioning: bucket.versioning,
+          storageType: bucket.storageType,
+          bucketLayout: bucket.bucketLayout,
+          creationTime: bucket.creationTime,
+          modificationTime: bucket.modificationTime,
+          sourceVolume: bucket.sourceVolume,
+          sourceBucket: bucket.sourceBucket,
+          usedBytes: bucket.usedBytes,
+          usedNamespace: bucket.usedNamespace,
+          quotaInBytes: bucket.quotaInBytes,
+          quotaInNamespace: bucket.quotaInNamespace,
+          owner: bucket.owner,
+          acls: bucket.acls
+        };
+      });
+
+      const volumeBucketMap: Map<string, Set<Bucket>> = getVolumeBucketMap(dataSource);
+
+      // Set options for volume selection dropdown
+      const volumeOptions: Option[] = Array.from(
+        volumeBucketMap.keys()
+      ).map(k => ({
+        label: k,
+        value: k
+      }));
+
+      setLoading(false);
+
+      setSelectedVolumes((prevState) => {
+        if (prevState.length === 0) return volumeOptions;
+        return prevState;
+      });
+
+      setState({
+        ...state,
+        totalCount: totalCount,
+        volumeBucketMap: volumeBucketMap,
+        volumeOptions: volumeOptions,
+        lastUpdated: Number(moment())
+      });
+    }).catch(error => {
+      setLoading(false);
+      showDataFetchError(error.toString());
+    });
+  }
+
+  let autoReloadHelper: AutoReloadHelper = new AutoReloadHelper(loadData);
+
+  useEffect(() => {
+    autoReloadHelper.startPolling();
+    addAclColumn();
+    const initialVolume = getVolumeSearchParam();
+    if (initialVolume) {
+      setSelectedVolumes([{
+        label: initialVolume,
+        value: initialVolume
+      }]);
+    }
+    loadData();
+
+
```

Comment:
I think since the React docs already specifies this, it is unnecessary.
Ref: https://react.dev/reference/react/useEffect
```
setup: The function with your Effect’s logic. Your setup function may also optionally return a cleanup function.
When your component is added to the DOM, React will run your setup function.
After every re-render with changed dependencies, React will first run the cleanup function (if you provided it)
with the old values, and then run your setup function with the new values.
After your component is removed from the DOM, React will run your cleanup function.
```